src/App.js

```diff
@@ -1,74 +1,107 @@
-    form: false,
-    popup: false,
-    addbtn: true,
-  });
-  const [postId, setPostId] = useState(null);
+    const [todoItems, setTodoItems] = useState(
+        JSON.parse(localStorage.getItem('Todos')) || []
+    );
+    const [visible, setVisible] = useState({
+        form: false,
+        popup: false,
+        addbtn: true,
+    });
+    const [postId, setPostId] = useState(null);
 
-  const addTodo = (todo) => {
-    if (!todo.text.trim()) return;
-    todo.id = todoItems.length + 1;
-    const tmpArr = [...todoItems, todo];
-    setTodoItems(tmpArr);
-  };
+    const addTodo = (todo) => {
+        if (!todo.text.trim()) return;
+        todo.id = todoItems.length + 1;
+        const tmpArr = [...todoItems, todo];
+        localStorage.setItem('Todos', JSON.stringify(tmpArr));
+        setTodoItems(tmpArr);
+    };
 
-  const toggleActivity = (obj) => {
-    let tmpObj = { ...activeClass, ...obj };
-    setActiveClass(tmpObj);
-  };
+    const toggleVisability = (obj) => {
+        const tmpObj = {
+            ...visible, ...obj 
+        };
+        setVisible(tmpObj);
+    };
 
-  const setId = (id) => setPostId(id);
+    const setId = (id) => setPostId(id);
 
-  const onDelete = (id) => {
-    let tmpArr = [...todoItems];
-    tmpArr = tmpArr.filter((e) => e.id !== id);
-    tmpArr = reCalculateId(tmpArr);
-    setTodoItems(tmpArr);
-  };
+    const onDelete = (id) => {
+        let tmpArr = [...todoItems];
+        tmpArr = tmpArr.filter((e) => e.id !== id);
+        tmpArr = reCalculateId(tmpArr);
+        localStorage.setItem('Todos', JSON.stringify(tmpArr));
+        setTodoItems(tmpArr);
+    };
 
-  const reCalculateId = (arr) => {
-    let tmpArr = arr.map((e, i) => {
-      e.id = i + 1;
-      return e;
-    });
-    return tmpArr;
-  };
+    const reCalculateId = (arr) => {
+        const tmpArr = arr.map((e, i) => {
+            e.id = i + 1;
+            return e;
+        });
+        return tmpArr;
+    };
+
+    const onClickDeleteButton = () => {
+        onDelete(postId);
+        toggleVisability({
+            popup: false,
+        });
+    };
+
+    const onClickCrossButton = (id) => {
+        toggleVisability({
+            popup: true 
+        });
+        setId(id);
+    };
 
-  return (
-    <div className="App">
-      <Popup
-        onDelete={() => onDelete(postId)}
-        toggleActivity={toggleActivity}
-        activeClass={activeClass.popup}
-      />
-      <AddButton
-        activeClass={activeClass.addbtn}
-        toggleActivity={() => toggleActivity({ addbtn: false, form: true })}
-      />
-      <AddForm
-        activeClass={activeClass.form}
-        toggleActivity={toggleActivity}
-        addTodo={addTodo}
-      />
-      <TodoCards
-        setId={setId}
-        toggleActivity={toggleActivity}
-        todos={todoItems}
-      />
-    </div>
-  );
+    return (
+        <div className="App">
+            <Popup
+                onClickDeleteButton={onClickDeleteButton}
```

Comment:
If you want to use generic ```Popup``` you shouldn't use specific function names.
Let's imagine you have 10 different buttons for 10 different popups on your site. According to you solution, it will be 10 different props.
Moreover, you can have 2 different delete buttons and it should do different things. So in your case it will be something like `onClickDeleteButton` & `onClickAnotherDeleteButton`.

If you still want to use generic popup, it should contain universal props or receive a children as property.
Like this
```
<Popup>
   <YourComponent />
<Popup/>
```

or it can be `onCancel` and `onAccept` functions